homeassistant/components/sleepiq/config_flow.py

```diff
@@ -75,6 +79,41 @@ async def async_step_user(
             last_step=True,
         )
 
+    async def async_step_reauth(self, user_input: dict[str, Any]) -> FlowResult:
+        """Perform reauth upon an API authentication error."""
+        self._username = user_input[CONF_USERNAME]
+        return await self.async_step_reauth_confirm(user_input)
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Confirm reauth."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            data = {
+                CONF_USERNAME: self._username,
+                CONF_PASSWORD: user_input[CONF_PASSWORD],
+            }
+
+            if not (error := await try_connection(self.hass, data)):
+                assert self._username is not None
+                entry = await self.async_set_unique_id(self._username.lower())
```

Comment:
Was it possible for the username to not be lowercased since we get it from the config entry?

Reply:
The `user_input[CONF_USERNAME]` allows mixed case (used in the description as well) but the unique_id is consistently being set to lower case in the other config flow steps